### PR TITLE
Rsync deps updated/rebuilt with xz packaging.

### PR DESCRIPTION
**Pull request by satmandu**
- Done in service of getting rsync working with https://github.com/chromebrew/chromebrew/pull/7297
- Also Add self check section to rsync to catch issues on build.

Works properly:
- [x] `x86_64`
- [x] `i686`
- [x] `armv7l`

### Run the following to get this pull request's changes locally for testing.
```
CREW_TESTING_REPO=https://github.com/satmandu/chromebrew.git CREW_TESTING_BRANCH=rsync_deps CREW_TESTING=1 crew update
```

### CODE REVIEW

File: packages/popt.rb [hunk 1]
```diff
@@ -10,21 +10,22 @@ class Popt < Package
...
 
   binary_url({
-    aarch64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/popt/1.18-7182e46-1_armv7l/popt-1.18-7182e46-1-chromeos-armv7l.tar.zst',
-     armv7l: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/popt/1.18-7182e46-1_armv7l/popt-1.18-7182e46-1-chromeos-armv7l.tar.zst',
-       i686: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/popt/1.18-7182e46-1_i686/popt-1.18-7182e46-1-chromeos-i686.tar.zst',
-     x86_64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/popt/1.18-7182e46-1_x86_64/popt-1.18-7182e46-1-chromeos-x86_64.tar.zst'
+    aarch64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/popt/1.18-7182e46-1_armv7l/popt-1.18-7182e46-1-chromeos-armv7l.tar.xz',
+     armv7l: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/popt/1.18-7182e46-1_armv7l/popt-1.18-7182e46-1-chromeos-armv7l.tar.xz',
+       i686: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/popt/1.18-7182e46-1_i686/popt-1.18-7182e46-1-chromeos-i686.tar.xz',
+     x86_64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/popt/1.18-7182e46-1_x86_64/popt-1.18-7182e46-1-chromeos-x86_64.tar.xz'
```
Comment (uberhacker) [inline packages/popt.rb:16]: No zst like the rest?

File: packages/rsync.rb [hunk 2]
```diff
@@ -28,6 +28,7 @@ class Rsync < Package
   depends_on 'lz4' # R
   depends_on 'openssl' # R
   depends_on 'popt' # R
+  depends_on 'py3_cmarkgfm' => :build
```
Comment (uberhacker) [inline packages/rsync.rb:31]: Would this new dependency affect the existing binaries?

**Review by uberhacker — CHANGES_REQUESTED** [review]
See comments.

File: packages/rsync.rb [hunk 3]
```diff
@@ -38,6 +39,10 @@ def self.build
     system 'make'
   end
 
+  def self.check
+        system 'make check'
```
Comment (uberhacker) [inline packages/rsync.rb:43]:
```
packages/rsync.rb:43:3: C: [Correctable] Layout/IndentationWidth: Use 2 (not 6) spaces for indentation.
        system 'make check'
  ^^^^^^
```

File: packages/rsync.rb (hunk 2 above)
Comment (satmandu) [reply packages/rsync.rb:31]: It's only a build dependency to allow man pages to be generated correctly.

File: packages/rsync.rb (hunk 3 above)
Comment (satmandu) [reply packages/rsync.rb]: Fixed!

File: packages/rsync.rb (hunk 2 above)
Comment (uberhacker) [reply packages/rsync.rb:31]: Are man pages for the binaries generated correctly?

**Review by uberhacker — APPROVED** [review]

**Pull request merged**